processor/src/lib.rs

```diff
@@ -253,6 +253,18 @@ where
             return Err(ExecutionError::ProgramAlreadyExecuted);
         }
 
+        // Load the program's advice data into the advice provider
+        for (digest, values) in program.mast_forest().advice_map().clone().into_iter() {
+            if self.host.borrow().advice_provider().get_mapped_values(&digest).is_some() {
```

Comment:
Similarly here, could be return an error only if the values are different at the same key?

Reply:
Done.